haystack/document_stores/faiss.py

```diff
@@ -367,6 +369,8 @@ def update_embeddings(
                 if self.similarity == "cosine":
                     self.normalize_embedding(embeddings)
 
+                if not self.faiss_indexes[index].is_trained:
+                    self.faiss_indexes[index].train(embeddings)
```

Comment:
Since we train the index in `write_documents`, do we need to do it here as well? In other words, is there any case where `self.faiss_indexes[index].is_trained:` goes to `False` _after_ writing the docs into the document store that we need to call `train()` again?

Reply:
> Since we train the index in `write_documents`, do we need to do it here as well? In other words, is there any case where `self.faiss_indexes[index].is_trained:` goes to `False` _after_ writing the docs into the document store that we need to call `train()` again?

Well, in some cases, we do not add embeddings to documents while writing docs when we call 'write_documents' function, then the add_vectors in this function is "False", and hence we will not have chance to call self.faiss_indexes[index].train(embeddings). In these cases, when we use IVF* indexing, we will meet `self.faiss_indexes[index].is_trained:` goes to `False` after we call 'write_documents'